packages/opentelemetry-core/test/node/environment.test.ts

```diff
@@ -0,0 +1,198 @@
+import { diag } from '@opentelemetry/api';
+
+describe('environment utility functions', function () {
+  describe('getStringFromEnv', function () {
+    afterEach(function () {
+      delete process.env.FOO;
+    });
+
+    it('should treat empty string as undefined', function () {
+      process.env.FOO = '';
+      assert.strictEqual(getStringFromEnv('FOO'), undefined);
+      process.env.FOO = '   ';
+      assert.strictEqual(getStringFromEnv('FOO'), undefined);
+    });
+
+    it('should treat not defined as undefined', function () {
+      delete process.env.FOO;
+      assert.strictEqual(getStringFromEnv('FOO'), undefined);
+    });
+
+    it('should trim extra whitespace', function () {
+      process.env.FOO = 'test-string  ';
+      assert.strictEqual(getStringFromEnv('FOO'), 'test-string');
+      process.env.FOO = '  test-string';
+      assert.strictEqual(getStringFromEnv('FOO'), 'test-string');
+      process.env.FOO = '  test-string   ';
+      assert.strictEqual(getStringFromEnv('FOO'), 'test-string');
+    });
+
+    it('should extract string from env var', function () {
+      process.env.FOO = 'test-string';
+      assert.strictEqual(getStringFromEnv('FOO'), 'test-string');
+    });
+
+    it('should retain casing', function () {
+      process.env.FOO = 'TeSt StRINg';
+      assert.strictEqual(getStringFromEnv('FOO'), 'TeSt StRINg');
+    });
+  });
+
+  describe('getNumberFromEnv', function () {
+    afterEach(function () {
+      delete process.env.FOO;
+    });
+
+    it('should treat empty string as undefined', function () {
+      process.env.FOO = '';
+      assert.strictEqual(getNumberFromEnv('FOO'), undefined);
+      process.env.FOO = '   ';
+      assert.strictEqual(getNumberFromEnv('FOO'), undefined);
+    });
+
+    it('should treat not defined as undefined', function () {
+      delete process.env.FOO;
+      assert.strictEqual(getNumberFromEnv('FOO'), undefined);
+    });
+
+    it('should trim extra whitespace', function () {
+      process.env.FOO = '1.234  ';
+      assert.strictEqual(getNumberFromEnv('FOO'), 1.234);
+      process.env.FOO = '  1.234';
+      assert.strictEqual(getNumberFromEnv('FOO'), 1.234);
+      process.env.FOO = '  1.234   ';
+      assert.strictEqual(getNumberFromEnv('FOO'), 1.234);
+    });
+
+    it('should extract integer from env var', function () {
+      process.env.FOO = String(42);
+      assert.strictEqual(getNumberFromEnv('FOO'), 42);
+    });
+
+    it('should extract double from env var', function () {
+      process.env.FOO = String(1.234);
+      assert.strictEqual(getNumberFromEnv('FOO'), 1.234);
+    });
+
+    it('should extract infinity from env var', function () {
+      process.env.FOO = String(Infinity);
+      assert.strictEqual(getNumberFromEnv('FOO'), Infinity);
+    });
+
+    it('should treat NaN as undefined', function () {
+      process.env.FOO = String(NaN);
+      assert.strictEqual(getNumberFromEnv('FOO'), undefined);
+    });
```

Comment:
nit: Perhaps a test case that `getNumberFromEnv` of some bogus non-number value, like `'bar'`, results in undefined.

Reply:
good point, added in https://github.com/open-telemetry/opentelemetry-js/pull/5443/commits/ffe7e1e4389b5086bd4a4952d51b0ba4b6c0038e